src/Elements/WrapperStop.php

```diff
@@ -124,11 +127,36 @@ protected function compile()
 			$this->Template->parameters       = $arrParams;
 			$this->Template->sliderId         = 'swiper-' . $objContent->id; // unique name for an entry in the sliderConfig-variable
 
+			// check if the scripts should be combined
+			$combine = '';
+
+			// ptable for swiperStop is "tl_article"
+			// get the article-model for the content-model
+			$article = ArticleModel::findById((int)$objContent->pid);
+			if ($article) {
+				// get the page-model of the article-model
+				$page = PageModel::findById((int)$article->pid);
+
+				// if the page itself has no layout, check for a parent-page
+				while ($page && !$page->includeLayout) {
+					$page = PageModel::findById($page->pid);
+				}
+				if ($page && $page->layout) {
+					$layout = LayoutModel::findById((int)$page->layout);
+					// get the current layout-model of the page
+					if ($layout) {
+						if ($layout->add_swiper_scripts) {
+							$combine = '|static';
+						}
+					}
+				}
+			}
```

Comment:
You do not need all that ;). You can get the current page object with
```php
global $objPage;
```
or simply
```php
$GLOBALS['objPage']
```

Reply:
Is that okay or is there a "magic" symfony-way to get the current PageModel?

Reply:
I think there is currently no way to inject the current page object.

Reply:
Ah, actually the current page model is a request attribute under `pageModel`.

Reply:
What do you mean with:
>Ah, actually the current page model is a request attribute under `pageModel`.

Reply:
If you get the request e.g. via
```php
$request = System::getContainer()->get('request_stack')->getCurrentRequest()
```
(or via depdendency injection) you can get the page model of the current page via
```php
$request->attributes->get('pageModel');
```

Reply:
now I am using the request to get the pageModel. If no page could be found (I think the attribute is not available in 4.4), I use the `GLOBALS['objPage']`.
Thank you for pointing me to this direction. :)